__tests__/index.spec.js

```diff
@@ -214,11 +241,88 @@ describe('string-based items with output fn', () => {
     expect(rta.find('.rta__entity')).toHaveLength(0);
   });
 
-  it('should invoke onchange handler after selection', () => {
-    expect(mockedFn).toHaveBeenCalledTimes(2);
+  it('should invoke onChange handler after selection', () => {
+    expect(mockedChangeFn).toHaveBeenCalledTimes(2);
+  });
+
+  it('should invoke onCaretPositionChange handler after selection', () => {
+    expect(mockedCaretPositionChangeFn).toHaveBeenCalledTimes(2);
   });
 
   it('text in textarea should be changed', () => {
     expect(rta.find('textarea').node.value).toBe('__happy_face__some test :a');
   });
 });
+
+describe('using ref to the ReactTextareaAutocomplete to call methods', () => {
+  const mockedChangeFn = jest.fn();
+
+  class ReactTextareaAutocompleteWrapper extends Component {
+
+    getCaretPosition() {
+      return this.textareaRef.getCaretPosition();
+    }
+
+    setCaretPosition(position) {
+      this.textareaRef.setCaretPosition(position);
+    }
+
+    render() {
+      return (
+        <ReactTextareaAutocomplete
+          placeholder={'Write a message.'}
+          value={'Controlled text'}
+          onChange={mockedChangeFn}
+          className={'ownClassName'}
+          style={{ background: 'red' }}
+          loadingComponent={Loading}
+          trigger={{
+            ':': {
+              dataProvider: () => Promise.resolve(['happy_face', 'sad_face']),
+              component: SmileItemComponent,
+            },
+          }}
+          ref={(ref) => { this.textareaRef = ref; }}
+        />
+      );
+    }
+  }
+
+  const rtaWrapper = mount(<ReactTextareaAutocompleteWrapper />);
+  const rtaWrapperRef = rtaWrapper.instance();
+
+  it('should get the correct caret position initially', () => {
+    const actual = rtaWrapperRef.getCaretPosition();
+    const expected = 0;
+
+    expect(actual).toBe(expected);
+  });
+
+  xit('should get the correct caret position after typing in some data', () => {
```

Comment:
Let's remove this test since it's skipped, we could easily handle this with Cypress. 

If you want you could do it in separate PR or I will do it on my own. It's uppon to you. But I don't want to complicate this PR so much.

Reply:
Okay will remove that. Would love to work on another PR with this cypress integration :)

Reply:
Super cool! So let's close first this one and then let's see. 🕺